### PR TITLE
subversion: revert to Python 2

### LINKED ISSUE #53193: Subversion's Python bindings fail in 1.13.0_3 or later for Python 2.7 and 3.8
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

## What you were trying to do (and why)

I'm trying to use the Subversion Python bindings installed with Brew, which I've successfully done for many years and many releases of Subversion provided by Homebrew, until recently. The Trac Project, for which I'm a developer, has some modules and a suite of tests that use the Subversion Python bindings.

Here is my `gist-logs`: https://gist.github.com/rjollos/9d21dc5316a6ab6e8650bd253cd9272a

## What happened (include command output)

Using Subversion Python bindings works prior to e073d0ba. In e073d0ba the dependency was changed to from Python@2 to Python@3.8. The Subversion bindings in 1.13 release are not compatible with Python 3. In Subversion 1.14 the bindings have been made compatible with Python 3 ([see release notes](https://subversion.apache.org/docs/release-notes/1.14.html#python3)). I've tested Subversion 1.14dev built from source on my OSX computer and the bindings work fine with 1.14dev. In fact the entire Test Suite for the Trac Project passes, as further proof.

So I think you can switch the dependency to Python@3.8 when 1.14 is released (looks like that will be soon), but for now I believe the brew formula needs to be switched back to have a dependency on Python@2. Details below.

<details>
  <summary>Command output</summary>

I add the libraries into a Python 2.7 virtualenv created using pyenv:
```
svn_python=$(brew --prefix subversion)/lib/svn-python
site_packages=$(find $(pyenv prefix)/ -name "site-packages" -print)
ln -sf "$svn_python/svn" "$site_packages/"
ln -sf "$svn_python/libsvn" "$site_packages/"
echo "$svn_python/libsvn" > "$site_packages/svn.pth"
```

Using 1.13.0_3 (e073d0ba) or later results in the following:

```
$ python --version
Python 2.7.17
$ python -c "from svn import core"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/rjollos/.pyenv/versions/trac-2.7/lib/python2.7/site-packages/svn/core.py", line 26, in <module>
    from libsvn.core import *
  File "/Users/rjollos/.pyenv/versions/trac-2.7/lib/python2.7/site-packages/libsvn/core.py", line 17, in <module>
    _core = swig_import_helper()
  File "/Users/rjollos/.pyenv/versions/trac-2.7/lib/python2.7/site-packages/libsvn/core.py", line 16, in swig_import_helper
    return importlib.import_module('_core')
  File "/Users/rjollos/.pyenv/versions/2.7.17/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
ImportError: dlopen(/usr/local/opt/subversion/lib/svn-python/libsvn/_core.so, 2): Symbol not found: _PyBytes_AsString
  Referenced from: /usr/local/Cellar/subversion/1.13.0_3/lib/libsvn_swig_py-1.0.dylib
  Expected in: flat namespace
 in /usr/local/Cellar/subversion/1.13.0_3/lib/libsvn_swig_py-1.0.dylib
```

Doing the same with a Python 3.8 virtualenv also fails:
```
trac-py382) trac-dev $ python --version
Python 3.8.2
(trac-py382) trac-dev $ python -c "from svn import core"
Traceback (most recent call last):
  File "/Users/rjollos/.pyenv/versions/3.8.2/envs/trac-py382/lib/python3.8/site-packages/libsvn/core.py", line 14, in swig_import_helper
    return importlib.import_module(mname)
  File "/Users/rjollos/.pyenv/versions/3.8.2/Python.framework/Versions/3.8/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 657, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 556, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 1101, in create_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
ImportError: dlopen(/Users/rjollos/.pyenv/versions/3.8.2/envs/trac-py382/lib/python3.8/site-packages/libsvn/_core.so, 2): Symbol not found: _PyFile_AsFile
  Referenced from: /usr/local/Cellar/subversion/1.13.0_3/lib/libsvn_swig_py-1.0.dylib
  Expected in: flat namespace
 in /usr/local/Cellar/subversion/1.13.0_3/lib/libsvn_swig_py-1.0.dylib

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/rjollos/.pyenv/versions/3.8.2/envs/trac-py382/lib/python3.8/site-packages/svn/core.py", line 26, in <module>
    from libsvn.core import *
  File "/Users/rjollos/.pyenv/versions/3.8.2/envs/trac-py382/lib/python3.8/site-packages/libsvn/core.py", line 17, in <module>
    _core = swig_import_helper()
  File "/Users/rjollos/.pyenv/versions/3.8.2/envs/trac-py382/lib/python3.8/site-packages/libsvn/core.py", line 16, in swig_import_helper
    return importlib.import_module('_core')
  File "/Users/rjollos/.pyenv/versions/3.8.2/Python.framework/Versions/3.8/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ImportError: dlopen(/usr/local/lib/svn-python/libsvn/_core.so, 2): Symbol not found: _PyFile_AsFile
  Referenced from: /usr/local/Cellar/subversion/1.13.0_3/lib/libsvn_swig_py-1.0.dylib
  Expected in: flat namespace
 in /usr/local/Cellar/subversion/1.13.0_3/lib/libsvn_swig_py-1.0.dylib
```

However, if I install the revision of the formulae that is the parent of e073d0ba, which is b309c2ef, 
```
$ brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/b309c2efabeb977aa97288efb85d7ee5ad2808b0/Formula/subversion.rb
$ brew pin subversion
```

Then `python -c "from svn import core"` succeeds and the entire Trac Project test suite passes.

</details>

## What you expected to happen

The following should product no errors:
```
$ python --version
Python 2.7.17
$ python -c "from svn import core"
```

## Step-by-step reproduction instructions (by running `brew install` commands)

```
$ brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/b309c2efabeb977aa97288efb85d7ee5ad2808b0/Formula/subversion.rb
```
results in usable Subversion Python bindings.

However,
```
$ brew install subversion
```
at present installs 1.13.0_4, for which the Subversion Python bindings fail.

### PR DESCRIPTION
**Pull request by iMichka**
Closes #53193

- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by claui** [comment]
I still get an error when I try to run the test commands given in #53193:

```bash
$ python --version
Python 2.7.16
$ python -c "from svn import core"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named svn
```

File: Formula/subversion.rb [hunk 1]
```diff
@@ -30,10 +30,12 @@ class Subversion < Formula
   # build against Homebrew versions of
   # gettext, lz4, perl, sqlite and utf8proc for consistency
   depends_on "gettext"
+  depends_on :macos # Due to Python 2
+  # See https://github.com/Homebrew/homebrew-core/issues/53193#issue-600482673
+  # Will work with Python 3.8 in subversion 1.14
```
Comment (Bo98) [inline Formula/subversion.rb:35]: Is it just the Python bindings that need Python 2? Couldn't they just be installed on macOS only?

File: Formula/subversion.rb [hunk 2]
```diff
@@ -54,7 +56,7 @@ class Subversion < Formula
   patch :DATA
 
   def install
-    ENV.prepend_path "PATH", Formula["python@3.8"].opt_libexec/"bin"
+    ENV.prepend_path "PATH", "/System/Library/Frameworks/Python.framework/Versions/2.7/bin"
```
Comment (MikeMcQuaid) [inline Formula/subversion.rb:59]: Should already be in the path and used?

File: Formula/subversion.rb (hunk 1 above)
Comment (iMichka) [reply Formula/subversion.rb]: Yes, I'll handle that case in linuxbrew when merging this, I'll revert back to Python 3 there.

I want to keep these lines (depends_on :macos # Due to Python 2) as it is easy to grep through the repo and find the Python 2 formulae.

File: Formula/subversion.rb (hunk 2 above)
Comment (iMichka) [reply Formula/subversion.rb:59]: As there is no `depends_on python@2 / uses_from_macos python@2` at all in the dependency list, this is not in the path. So it needs to be added.

Comment (MikeMcQuaid) [reply Formula/subversion.rb:59]: `/usr/bin/python` isn't in the PATH by default?

Comment (iMichka) [reply Formula/subversion.rb]: Right, my bad. I removed the line.

**Pull request closed without merging**